mypy/checker.py

```diff
@@ -606,6 +360,8 @@ def visit_func_def(self, defn: FuncDef) -> Type:
             else:
                 # Function definition overrides a variable initialized via assignment.
                 orig_type = defn.original_def.type
+                # XXX This can be None, as happens in
+                # test_testcheck_TypeCheckSuite.testRedefinedFunctionInTryWithElse
```

Comment:
I set the test to -skip.  It wasn't obvious to me how to fix it, and I don't think the issue is introduced by this PR -- it's that the old binder had bugs so it didn't check the test's branches fully.